tensorflow/compiler/xla/service/llvm_ir/math_ops.cc

```diff
@@ -14,13 +14,19 @@ limitations under the License.
 ==============================================================================*/
 
 #include "tensorflow/compiler/xla/service/llvm_ir/math_ops.h"
+
 #include "tensorflow/compiler/xla/service/llvm_ir/llvm_util.h"
 
 namespace xla {
 namespace llvm_ir {
 
-llvm::Value* EmitFastTanh(llvm::IRBuilder<>* b, llvm::Value* input) {
+llvm::Value* EmitFastTanh(llvm::IRBuilder<>* b, llvm::Value* input,
+                          const TanhType input_type /* = TanhType::Double*/) {
   llvm::Type* type = input->getType();
+  const float plus_clamp =
+      (input_type == TanhType::Double) ? 9.0 : 7.99881172180175781f;
+  const float minus_clamp =
+      (input_type == TanhType::Double) ? -9.0 : -7.99881172180175781f;
```

Comment:
Move out constant into a `static constexpr`, reuse it for positive/negative case and give it a name? Also maybe a short comment on what values are chosen and why?

Reply:
Origin version is 9.0, and I don't know why.